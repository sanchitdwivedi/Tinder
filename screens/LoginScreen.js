import { View, Text, Button } from 'react-native'
import React from 'react'
import useAuth from '../hooks/useAuth'

const LoginScreen = () => {
    const {signInWithGoogle} = useAuth();
  return (
    <View>
      <Text>Login to the app</Text>
      <Button title="Login" onPress={signInWithGoogle} />
    </View>
  )
}

export default LoginScreen