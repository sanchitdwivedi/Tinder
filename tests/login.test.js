import React from 'react';
import renderer from 'react-test-renderer'
import LoginScreen from '../screens/LoginScreen'

describe('Login', () => {
    describe('Rendering', () => {
        it("renders login screen without crash", () => {
            const login = renderer.create(<LoginScreen/>).toJSON();
            expect(login).toMatchSnapshot();
        })
    });
});