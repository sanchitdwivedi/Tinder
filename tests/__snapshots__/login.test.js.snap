// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Rendering renders login screen without crash 1`] = `
<View
  style={
    Object {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 16,
          "bottom": 160,
          "marginHorizontal": "25%",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "absolute",
          "width": 208,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Sign in & get swiping
      </Text>
    </View>
  </View>
</View>
`;
